model/src/main/java/org/cbioportal/model/summary/MutationSummary.java

```diff
@@ -0,0 +1,259 @@
+package org.cbioportal.model.summary;
+
+import java.io.Serializable;
+
+public abstract class MutationSummary implements Serializable {
```

Comment:
Hey Ersin, sorry I missed that response of yours somehow!

I would contend that while the domain logic is not overly complex here indeed, it's not trivial either (for instance consider the EAV-like `clinical_attribute` table). I would really recommend shielding a core domain logic from that kind of low-level persistence considerations*. Right now these transaction scripts are highly coupled and display poor cohesion because they need to worry about both duties (though I only looked at a small subset of the codebase, so this is to be taken with a grain of salt).

I'm not familiar with mapstruct, but I know dozer which I believe plays a similar role. This tools help indeed, though really they just compensate for the weaknesses of the language. Indeed Java beans are notoriously painful to transform. I would however not let that stand in the way having a properly layered architecture... Admittedly these are the kind of issues that made me move away from java altogether though.

Just my two cents :)

*: IMO this is really more indicative of the need to use relation/document-store hybrid like postgres. Jumping into a full-blown document store like mongo would be overkill however, as the input file format is very relational in nature.